aws-logs-subscriptionfilter/src/main/java/software/amazon/logs/subscriptionfilter/CreateHandler.java

```diff
@@ -0,0 +1,66 @@
+package software.amazon.logs.subscriptionfilter;
+
+import com.amazonaws.util.StringUtils;
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClient;
+import software.amazon.awssdk.services.cloudwatchlogs.model.*;
+import software.amazon.cloudformation.exceptions.*;
+import software.amazon.cloudformation.proxy.*;
+import software.amazon.cloudformation.resource.IdentifierUtils;
+
+public class CreateHandler extends BaseHandlerStd {
+    private Logger logger;
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<CloudWatchLogsClient> proxyClient,
+            final Logger logger) {
+        this.logger = logger;
+        final ResourceModel model = request.getDesiredResourceState();
+
+        if (StringUtils.isNullOrEmpty(model.getFilterName())) {
+            final String resourceIdentifier = IdentifierUtils.generateResourceIdentifier(request.getLogicalResourceIdentifier(), request.getClientRequestToken());
```

Comment:
I believe in Native, the generated resourceId/Filter Name seems to be stackname + logicalresourceId + UUID . There is also a limit in native PHYSICAL_RESOURCE_ID_MAX_LENGTH = 512 which you can pass. 
https://github.com/aws-cloudformation/cloudformation-cli-java-plugin/blob/master/src/main/java/software/amazon/cloudformation/resource/IdentifierUtils.java